### PR TITLE
chore(docs): initial-ssp-generate-doc

### LINKED ISSUE #275: Document: System-Security-Plan generation
Research and documentation for how Lula will generate and operate on a `system-security-plan`. 

## Objective 
Establish a document for `system-security-plan` that evolves as research develops around:
- The purpose of a `system-security-plan` generically
- `component-definition` mapping to `system-security-plan`
- How Lula performs this mapping during generation
- Expectations for hybrid (automation/manual) authorship of the artifact

Consider this a research spike that also educates/informs others about opinionation of Lula workflows. Human readable format means a lower barrier to entry for external expertise to evaluate and challenge assumptions.

### PR DESCRIPTION
**Pull request by CloudBeard**
## Description

This ssp generate document also has a digram in Miro that helps illustrate the depency graph.

! Due to the size of OSCAL Models including a Mermaid diagram is illegible.

This doc outlines the elements contained within the component-definition and profile that help fill out the ssp. 

It also contains the items that are needed through a config file and the elements that need augmented within the component-definition.

## Related Issue

Relates to https://github.com/orgs/defenseunicorns/projects/12/views/5?pane=issue&itemId=54551768

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [ ] [Schema Updates](https://github.com/defenseunicorns/lula/blob/main/docs/community-and-contribution/schema-updates.md) applied
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/lula/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Comment by brandtkeller** [comment]
For posterity - discussed splitting the content into `docs/oscal/system-security-plan.md` for content relating to the model generically and `cli-commands/generate/system-security-plan.md` for the logical implementation documentation.

File: docs/cli-commands/generation/system-security-plan.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# System Security Plan
```
Comment (brandtkeller) [inline docs/cli-commands/generation/system-security-plan.md:1]: All docs in theory will be published to our website. I would request a note of some type at the top of this document to mention that this is proposed work captured for research purposes.

File: docs/cli-commands/generation/system-security-plan.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+- The `LulaOscalConfig` file `--config`
+
+The following command will generate a system security plan with the above context:
+```
+lula generate system-security-plan --component .src/test/unit/valid-component.yaml -c https://raw.githubusercontent.com/usnistgov/oscal-content/main/nist.gov/SP800-53/rev5/json/NIST_SP-800-53_rev5_HIGH-baseline-resolved-profile_catalog.json --config ./LulaOscalConfig
+```
+
+There are optional flags that can be added to the command to generate a system security plan:
+
+- The output file of the component `-o` or `--output`; `oscal-component.yaml`
```
Comment (brandtkeller) [inline docs/cli-commands/generation/system-security-plan.md]: `output` would be an ssp not component yes?

File: docs/cli-commands/generation/system-security-plan.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+For the items in `Elements in SSP Not in Component Definition` a "config" file will be needed to fill in the gaps. There are also metadata fields such as `responsible-roles`, `responsible-parties`, and `parties` that can be added to the `system-security-plan` through the config file that may not be necessary to add directly to the `component-definition`
```
Comment (brandtkeller) [inline docs/cli-commands/generation/system-security-plan.md]: We need to generalize more before we can definitively say what the path forward will be. IE "context will be needed to fill in the gaps as well as establish common data across models".

File: docs/cli-commands/generation/system-security-plan.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+### Elements in Component Definition Not in SSP
```
Comment (brandtkeller) [inline docs/cli-commands/generation/system-security-plan.md:36]: Of the categories below - am I missing "Applicable fields in Component Definition that translate to an SSP" somewhere?

File: docs/oscal/system-security-plan.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# System Security Plan
+
+A [System Security Plan](https://pages.nist.gov/OSCAL-Reference/models/v1.1.2/system-security-plan/json-reference/#/system-security-plan) is an OSCAL-specific model to represent a system as a whole. In Lula, the `generate system-security-plan` command creates an `oscal-system-security-plan` object to explain the system as a whole by using the compliance data provided by the `component-definition` and `LulaOscalConfig`.
```
Comment (brandtkeller) [inline docs/oscal/system-security-plan.md]: Let's drop mention of `LulaOscalConfig` - we know we want to implement generate functionality - but templating and variables is still a research topic.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Would like to generalize some of the content to remove mention of topics that still have a large amount of discovery to perform. 

These docs will go live on our docs site upon renovate merge to the `lula-docs` repo - we should be explicit in defining the `generation` document as proposed work and not yet implemented.

File: docs/cli-commands/generation/system-security-plan.md (hunk 1 above)
Comment (meganwolf0) [reply docs/cli-commands/generation/system-security-plan.md:1]: So I think with that header, the menu item will be "**NOTE:** This Document is in an active research phase." - wondering if instead we might want like a future or proposed or something directory? Also, wondering if we did like a .dir if that would be hidden from rendering in the website... possibly a question to @bdfinst

Comment (brandtkeller) [reply docs/cli-commands/generation/system-security-plan.md:1]: agree on moving the `note` below the main H1 at a minimum. 

Open to thoughts on future content. I like the idea of ensuring they get exposed in our repo/website. We may be constrained by the amount of features we can develop currently - but getting general intent published will hopefully go a long way by generating some demand signals.

**Review by meganwolf0 — DISMISSED** [review]

File: docs/oscal/assessment-results.md [hunk 6]
```diff
@@ -35,4 +35,5 @@ The primary structure for Lula production and operation of `assessment-results`
 - Results are sorted by `start` time in descending order
 - Findings are sorted by `target.target-id` in ascending order
 - Observations are sorted by `collected` time in ascending order
-- Back Matter Resources are sorted by `title` in ascending order.
\ No newline at end of file
+- Back Matter Resources are sorted by `title` in ascending order.
+- Back Matter Resources are sorted by `title` in ascending order.
```
Comment (brandtkeller) [inline docs/oscal/assessment-results.md]: Why are there duplicates of this line item?

**Review by brandtkeller — APPROVED** [review]

**Pull request merged**